rocAL/rocAL/source/meta_data/meta_node_rotate.cpp

```diff
@@ -59,40 +54,47 @@ void RotateMetaNode::update_parameters(pMetaDataBatch input_meta_data, pMetaData
             BoundingBoxCord box;
             float src_bb_x, src_bb_y, bb_w, bb_h;
             float dest_cx, dest_cy, src_cx, src_cy;
-            float x1, y1, x2, y2, x3, y3, x4, y4, min_x, min_y;
+            float x1, y1, x2, y2, x3, y3, x4, y4, min_x, min_y, max_x, max_y;
             float rotate[4];
             float radian = RAD(_angle_val[i]);
             rotate[0] = rotate[3] = cos(radian);
             rotate[1] = sin(radian);
             rotate[2] = -1 * rotate[1];
             dest_cx = _dst_width / 2;
             dest_cy = _dst_height / 2;
-            src_cx = _src_width_val[i]/2;
-            src_cy = _src_height_val[i]/2;
+
+            src_cx = static_cast<float>(input_roi[i].x2) / 2;
+            src_cy = static_cast<float>(input_roi[i].y2) / 2;
             src_bb_x = coords_buf[j].l;
             src_bb_y = coords_buf[j].t;
-            bb_w = coords_buf[j].r;
-            bb_h = coords_buf[j].b;
+            bb_w = coords_buf[j].r - coords_buf[j].l;
+            bb_h = coords_buf[j].b - coords_buf[j].t;
             x1 = (rotate[0] * (src_bb_x - src_cx)) + (rotate[1] * (src_bb_y - src_cy)) + dest_cx;
             y1 = (rotate[2] * (src_bb_x - src_cx)) + (rotate[3] * (src_bb_y - src_cy)) + dest_cy;
-            x2 = (rotate[0] * ((src_bb_x + bb_w) - src_cx))+( rotate[1] * (src_bb_y - src_cy)) + dest_cx;
-            y2 = (rotate[2] * ((src_bb_x + bb_w) - src_cx))+( rotate[3] * (src_bb_y - src_cy)) + dest_cy;
+            x2 = (rotate[0] * ((src_bb_x + bb_w) - src_cx)) + ( rotate[1] * (src_bb_y - src_cy)) + dest_cx;
+            y2 = (rotate[2] * ((src_bb_x + bb_w) - src_cx)) + ( rotate[3] * (src_bb_y - src_cy)) + dest_cy;
             x3 = (rotate[0] * (src_bb_x - src_cx)) + (rotate[1] * ((src_bb_y + bb_h) - src_cy)) + dest_cx;
             y3 = (rotate[2] * (src_bb_x - src_cx)) + (rotate[3] * ((src_bb_y + bb_h) - src_cy)) + dest_cy;
-            x4 = (rotate[0] * ((src_bb_x + bb_w) - src_cx))+( rotate[1] * ((src_bb_y + bb_h) - src_cy)) + dest_cx;
-            y4 = (rotate[2] * ((src_bb_x + bb_w) - src_cx))+( rotate[3] * ((src_bb_y + bb_h) - src_cy)) + dest_cy;
+            x4 = (rotate[0] * ((src_bb_x + bb_w) - src_cx)) + ( rotate[1] * ((src_bb_y + bb_h) - src_cy)) + dest_cx;
+            y4 = (rotate[2] * ((src_bb_x + bb_w) - src_cx)) + ( rotate[3] * ((src_bb_y + bb_h) - src_cy)) + dest_cy;
             min_x = std::min(x1, std::min(x2, std::min(x3, x4)));
             min_y = std::min(y1, std::min(y2, std::min(y3, y4)));
-            box.l = std::max(min_x, 0.0f);
-            box.t = std::max(min_y, 0.0f);
-            box.r = std::max(x1, std::max(x2, std::max(x3, x4))); ;
-            box.b = std::max(y1, std::max(y2, std::max(y3, y4)));
+            max_x = std::max(x1, std::max(x2, std::max(x3, x4)));
+            max_y = std::max(y1, std::max(y2, std::max(y3, y4)));
+            box.l = std::min(0.0f, min_x);
+            box.t = std::min(0.0f, min_y);
+            box.r = max_x;
```

Comment:
Why do we need max_x and max_y variables additionally? We can directly update here right please check